src/Data/Fin/Induction.agda

```diff
@@ -59,6 +60,20 @@ open WF public using (Acc; acc)
   induct {suc i} (acc rec) = Pᵢ⇒Pᵢ₊₁ i (induct (rec (inject₁ i) i<i+1))
     where i<i+1 = ℕ<⇒inject₁< (i<1+i i)
 
+≤-Induction : ∀ (P : Pred (Fin (suc n)) ℓ) {i} →
+              P i →
+              (∀ {i} → P (inject₁ i) → P (suc i)) →
+              ∀ {j} → j ≥ i → P j
+≤-Induction P {i} Pi Pᵢ⇒Pᵢ₊₁ {j} j≥i = induct (<-wellFounded _) (<-cmp i j) j≥i
```

Comment:
Also we should be able to redefine `<-weakInduction` in terms of this right?

Reply:
Yes, it is now using it.